torch/_dynamo/eval_frame.py

```diff
@@ -691,138 +696,174 @@ def result_capturing_wrapper(*graph_inputs):
         return result_capturing_wrapper
 
     flat_args, in_spec = pytree.tree_flatten((args, kwargs))
+    # apply constraints to flat_args
+    # First, reconstruct constraints to be a map:[int] view
+    if not constraints:
+        constraints = list()
 
-    remove_from_cache(f)
-    with patch(f"{__name__}.most_recent_backend", None), config.patch(
-        specialize_int=True
-    ):
-        opt_f = optimize_assert(
-            dynamo_normalization_capturing_compiler,
-            hooks=Hooks(guard_export_fn=guard_export_print, guard_fail_fn=None),
-            export=True,
-            dynamic=(tracing_mode == "symbolic"),
-        )(f)
-        # TODO(voz): We may have instances of `f` that mutate inputs, we should track sideffects and reject.
-        result_traced = opt_f(*args, **kwargs)
-    remove_from_cache(f)
+    tensor_to_indices: Dict[torch.Tensor, Set[int]] = dict()
+    try:
+        for constraint in constraints:
+            tensor = constraint[0]
+            index = constraint[1]
+            # TODO(voz): Is tensor defaultdict friendly?
+            if tensor not in tensor_to_indices:
+                assert not torch._dynamo.has_dynamic_dims(
+                    tensor
+                ), "Illegal to export tensor already marked dynamic"
+                tensor_to_indices[tensor] = set()
+            assert index not in tensor_to_indices[tensor], f"Duplicate index {index}"
+            tensor_to_indices[tensor].add(index)
+            torch._dynamo.mark_dynamic(tensor, index)
```

Comment:
You're going to have to rewrite all this code when you add range constraints...

Reply:
Yes, but we don't have the mechanism for those yet, so I am fine with that for now.